### PR TITLE
specs: Exclusively place proxies for MEAN example

### PR DESCRIPTION
**Pull request by LuiseV**
With this update, we place haproxy containers on all Worker VMs in the MEAN app
example deployment. This allows us to access the web app via any Worker VM,
which simplifies the instructions in DeployMEANapp.md.

### CODE REVIEW

**Comment by tejasmanohar** [comment]
FWIW, I think the API for exclusive placement rules is unintuitive (e.g. `true`)

File: specs/mean/example.js [hunk 1]
```diff
@@ -18,7 +18,12 @@ namespace.deploy(baseMachine.asWorker().replicate(3));
 
 var mongo = new Mongo(3);
 var app = new App(3, 8080, { MONGO_URI: mongo.uri("mean-example") });
-var haproxy = new HaProxy(2, app.services(), 8080);
+var haproxy = new HaProxy(3, app.services(), 8080);
+
+// Places all haproxy containers on separate Worker VMs.
+// This is just for convenience for the example instructions, as it allows us to
+// access the web application by using the IP address of any Worker VM.
+haproxy.service.place(new LabelRule(true, haproxy.service));
```
Comment (tejasmanohar) [inline specs/mean/example.js]: Not sure I follow. HAP runs on 80 so can't you always use the public IP of any VM?

**Comment by LuiseV** [comment]
@tejasmanohar I think we all agree on that. Placement is going to go through a major redesign at some point when we know what the right design is.

File: specs/mean/example.js (hunk 1 above)
Comment (LuiseV) [inline specs/mean/example.js]: Nope, you can only access the website through VMs that have a HAP container

Comment (tejasmanohar) [inline specs/mean/example.js]: oh this makes it placed _everywhere_. hm, not sure it's a great default. i'd rather explain quilt containers & machines to the user

Comment (tejasmanohar) [inline specs/mean/example.js]: once floating IPs are in, it won't matter as much

Comment (LuiseV) [inline specs/mean/example.js]: Yeah, floating IPs should eliminate this step. Also, I agree that this is not a good production default, which is why I left the comment saying that this is just to simplify our guide. For the purpose of the brief introduction to Quilt, I think it is fine. Finding the right machine by using `quilt containers` is not intuitive right now, and I don't think it is crucial for the purpose of the getting started guide. As soon as floating IPs are in, I we should be able to scratch this :)

Comment (kklin) [inline specs/mean/example.js]: @LuiseV do you have any idea how we should improve `quilt containers`? It seems like we need some sort of unified view with the machine it's scheduled on, etc. Also prettier formatting and a better way to reference items than the database ID.
Might be something @stevenlin598 could do.

Comment (tejasmanohar) [inline specs/mean/example.js]: machine ID seems like a good addition

Comment (tejasmanohar) [inline specs/mean/example.js]: or maybe a "quilt status" that shows machines & containers in one place

**Pull request merged**